packages/core/src/routes/session.ts

```diff
@@ -264,13 +264,24 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
 
   router.post('/session/consent', async (ctx, next) => {
     const interaction = await provider.interactionDetails(ctx.req, ctx.res);
-    const { session, grantId, params, prompt } = interaction;
+    const {
+      session,
+      grantId,
+      params: { client_id },
+      prompt,
+    } = interaction;
     assertThat(session, 'session.not_found');
 
     const { accountId } = session;
     const grant =
       conditional(grantId && (await provider.Grant.find(grantId))) ??
-      new provider.Grant({ accountId, clientId: String(params.client_id) });
+      new provider.Grant({ accountId, clientId: String(client_id) });
+
+    // eslint-disable-next-line promise/prefer-await-to-then
+    void saveUserFirstConsentedAppId(accountId, String(client_id)).catch((error) => {
```

Comment:
nit: can we use the if statement directly in this place? seems no need to wrap this function.

Reply:
And why ignore errors? seems is different from logs.

Reply:
use the function name `saveUserFirstConsentedAppId ` to explain doing what.

Reply:
refactored. just `await saveUserFirstConsentedAppId(…`